test/Microsoft.AspNetCore.Razor.Tools.Test/RequestDispatcherTest.cs

```diff
@@ -0,0 +1,552 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CommandLine;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Tools
+{
+    public class RequestDispatcherTest
+    {
+        private static BuildRequest EmptyBuildRequest => new BuildRequest(1, Array.Empty<BuildRequest.Argument>());
+
+        private static BuildResponse EmptyBuildResponse => new CompletedBuildResponse(
+            returnCode: 0,
+            utf8output: false,
+            output: string.Empty);
+
+        [Fact]
+        public async Task ReadFailure()
```

Comment:
These need better names across the board. It's not really clear what's the expected outcome and why.

Reply:
A better name would be something like `Dispatcher_WhenReadingRequestFails_ClosesConnection`